pkg/sql/logictest/testdata/logic_test/select

```diff
@@ -719,3 +719,24 @@ query BI
 SELECT * FROM t44132 WHERE c0
 ----
 true NULL
+
+# Tests for `disallow_full_table_scans`
+statement ok
+CREATE TABLE t_disallow_scans(a INT);
+
+statement ok
+INSERT INTO t_disallow_scans VALUES (1), (2), (3)
+
+statement ok
+SELECT * FROM t_disallow_scans
+
+statement ok
+SET disallow_full_table_scans = true
+
+statement error pq: query `SELECT \* FROM t_disallow_scans` contains a full table/index scan which is explicitly disallowed
```

Comment:
Add a test for a statement that causes an internal executor to perform a full scan (like a virtual table using an IE). Ensure that that works.